### PR TITLE
Transform labels into sequences

### PR DESCRIPTION
**Pull request by m09**
This is a heavy one :/

Signed-off-by: Hugo Mougard <hugo@sourced.tech>

### CODE REVIEW

**Comment by m09** [comment]
webhooks seem completely fucked up. The travis build succeeded and github is not aware.

File: lookout/style/format/rules.py [hunk 1]
```diff
@@ -112,7 +113,8 @@ def apply(self, X: numpy.ndarray, return_winner_indices=False
         return prediction
 
     def predict(self, X: numpy.ndarray, vnodes_y: Sequence[VirtualNode],
-                vnodes: Sequence[VirtualNode], language: str, return_originals: bool = False
+                vnodes: Sequence[VirtualNode], language: str, feature_extractor: FeatureExtractor,
```
Comment (vmarkovtsev) [inline lookout/style/format/rules.py]: We can take `language` from `feature_extractor`. And actually `language` should be completely abstracted from the code here and in `postprocess` and all the language-dependent functions should be consumed from `feature_extractor`

File: lookout/style/format/langs/javascript/postprocessor.py [hunk 2]
```diff
@@ -21,32 +50,42 @@ def postprocess(X: numpy.ndarray, y: numpy.ndarray, vnodes_y: Sequence[VirtualNo
...
-            winner, loser = ((y_i, y_i_paired)
-                             if conf_vnode > conf_vnode_paired
-                             else (y_i_paired, y_i))
-            processed_y[loser] = y[winner]
-            processed_winners[loser] = winners[winner]
+        y_i_1 = y_indices[id(vnode1)]
+        y_i_3 = y_indices[id(vnode3)]
+        conf_vnode1 = rules.rules[winners[y_i_1]].stats.conf
+        conf_vnode3 = rules.rules[winners[y_i_3]].stats.conf
+        labels1 = tuple(feature_extractor.composite_to_labels[y[y_i_1]])
```
Comment (vmarkovtsev) [inline lookout/style/format/langs/javascript/postprocessor.py]: We can make them lists and avoid the conversions below. At least I don't see why not.

**Comment by vmarkovtsev** [comment]
This is a tough one :) I looked it through, of course, could not review the core meaning - but that's not really required now.

**Pull request merged**